includes/class-migrator-cli-utils.php

```diff
@@ -101,20 +101,115 @@ public static function rest_request( $endpoint, $body = array() ) {
 	/**
 	 * Executes a request to the Shopify GraphQL API
 	 *
-	 * @param array $body the request body.
-	 * @return array
+	 * @param string $query The GraphQL query string.
+	 * @param array  $variables Optional. Variables for the query.
+	 * @return object|WP_Error Decoded data object on success, WP_Error on failure.
 	 */
-	public static function graphql_request( $body ) {
-		return wp_remote_post(
-			'https://' . SHOPIFY_DOMAIN . '/admin/api/2023-04/graphql.json',
-			array(
-				'headers' => array(
-					'X-Shopify-Access-Token' => ACCESS_TOKEN,
-					'Content-Type'           => 'application/json',
-				),
-				'body'    => wp_json_encode( $body ),
-			)
+	public static function graphql_request( $query, $variables = array() ) {
+		$max_retries = 10;
+		$retry_count = 0;
+
+		$body = array(
+			'query' => $query,
+			'variables' => $variables,
 		);
+
+		while ( $retry_count <= $max_retries ) {
+			$response = wp_remote_post(
+				'https://' . SHOPIFY_DOMAIN . '/admin/api/2023-04/graphql.json',
```

Comment:
@naman03malhotra This API version doesn't seem to be supported any longer: https://shopify.dev/docs/api/release-notes/previous-versions/2023-04.

Have you tried using the latest API version here?